app/assets/javascripts/darkswarm/services/stripe_elements.js.coffee

```diff
@@ -34,26 +35,35 @@ Darkswarm.factory 'StripeElements', ($rootScope, Loading, RailsFlashLoader) ->
           Loading.clear()
           RailsFlashLoader.loadFlash({error: t("error") + ": #{response.error.message}"})
           @triggerAngularDigest()
+          console.error(JSON.stringify(response.error))
         else
           secrets.token = response.paymentMethod.id
-          secrets.cc_type = response.paymentMethod.card.brand
+          secrets.cc_type = @mapPaymentMethodsApiCardBrand(response.paymentMethod.card.brand)
           secrets.card = response.paymentMethod.card
           submit()
 
     triggerAngularDigest: ->
       # $evalAsync is improved way of triggering a digest without calling $apply
       $rootScope.$evalAsync()
 
-    # Maps the brand returned by Stripe to that required by activemerchant
-    mapCC: (ccType) ->
-      switch ccType
+    # Maps the brand returned by Stripe's tokenAPI to that required by activemerchant
+    mapTokenApiCardBrand: (cardBrand) ->
+      switch cardBrand
         when 'MasterCard' then return 'master'
         when 'Visa' then return 'visa'
         when 'American Express' then return 'american_express'
         when 'Discover' then return 'discover'
         when 'JCB' then return 'jcb'
         when 'Diners Club' then return 'diners_club'
 
+    # Maps the brand returned by Stripe's paymentMethodsAPI to that required by activemerchant
+    mapPaymentMethodsApiCardBrand: (cardBrand) ->
```

Comment:
I understand this PR pretty needed that is why I approve, but we'll need to remove this duplication right after.

Reply:
ah, I think we have discussed this before. we dont have a way to share code between darkswarm and admin!